torchbenchmark/util/backends/cudagraph.py

```diff
@@ -1,26 +1,37 @@
 import torch
-from typing import Tuple
+from torchbenchmark.util.backends import create_backend
+from typing import List
 
-def enable_cudagraph(model: 'torchbenchmark.util.model.BenchmarkModel', example_inputs: Tuple[torch.tensor]):
-    optimizer = model.optimizer
-    loss_fn = model.loss_fn
-    # warmup
-    s = torch.cuda.Stream()
-    s.wait_stream(torch.cuda.current_stream())
-    with torch.cuda.stream(s):
-        for _ in range(3):
-            optimizer.zero_grad(set_to_none=True)
-            y_pred = model.model(*example_inputs)
-            loss = loss_fn(y_pred, model.example_outputs)
-            loss.backward()
-            optimizer.step()
-    torch.cuda.current_stream().wait_stream(s)
-    # capture
-    g = torch.cuda.CUDAGraph()
-    optimizer.zero_grad(set_to_none=True)
-    with torch.cuda.graph(g):
-        static_y_pred = model.model(*example_inputs)
-        static_loss = loss_fn(static_y_pred, model.example_outputs)
-        static_loss.backward()
-        optimizer.step()
-    model.g = g
\ No newline at end of file
+WARMUP_ITER = 3
+
+@create_backend
+def cudagraph(model: 'torchbenchmark.util.model.BenchmarkModel', backend_args: List[str]):
+    cudagraph_func_name = f"cudagraph_{model.test}"
+    assert hasattr(model, cudagraph_func_name), f"CUDA Graph only works on models implement {cudagraph_func_name}()"
+    if model.test == "train":
+        assert hasattr(model, "SKIP_ZERO_GRAD"), f"The model must support skipping zero grad in its train test."
```

Comment:
should this be `getattr(model, "SKIP_ZERO_GRAD", False)` ? in case SKIP_ZERO_GRAD is defined and is set to false

Reply:
`model.SKIP_ZERO_GRAD` should be False in eager mode, we should only set it to True when running with cudagraph. So it is fine to be False here.